calculator_test.go

```diff
@@ -5,20 +5,82 @@ import (
 	"testing"
 )
 
-func TestAdd(t *testing.T) {
-	t.Parallel()
-	var want float64 = 4
-	got := calculator.Add(2, 2)
-	if want != got {
-		t.Errorf("want %f, got %f", want, got)
+func TestCalculator(t *testing.T) {
+
+	var tests = []struct {
+		function func(float64, float64) float64
+		want     float64
+		input    []float64
+	}{
+		{function: calculator.Add, input: []float64{2, 2}, want: 4},
+		{function: calculator.Subtract, input: []float64{2, 4}, want: 2},
+		{function: calculator.Multiply, input: []float64{2, 4}, want: 8},
+	}
+	for _, tt := range tests {
+		got := tt.function(tt.input[0], tt.input[1])
+		if tt.want != got {
+			t.Errorf("want %f, got %f", tt.want, got)
+		}
+
+	}
+}
+
+func TestDivide(t *testing.T) {
+	var tests = []struct {
+		a             float64
+		b             float64
+		want          float64
+		errorExpected bool
+	}{
+		{
+			a:             4,
+			b:             0,
+			want:          999,
+			errorExpected: true},
+		{
+			a:             4,
+			b:             2,
+			want:          2,
+			errorExpected: false,
+		},
+	}
+	for _, tt := range tests {
+		got, err := calculator.Divide(tt.a, tt.b)
+		if tt.errorExpected && err == nil {
+			t.Fatal("want err/0, got nil")
```

Comment:
Let's be more explicit:

```suggestion
			t.Fatal("want err dividing by zero, got nil")
```

Helpful failure messages are one of the biggest things we can do to make our tests more valuable and useful. At some point the test _will_ fail (if not, there's no point having the test), and when that happens, it's a demotivating experience for any programmer. We want to guide them back to having passing tests as fast and as painlessly as possible.